File: app/src/main/java/com/quiz/mallzellij_show_android/model/AuthResponse.java
package com.quiz.mallzellij_show_android.model;

import com.google.gson.annotations.SerializedName;

public class AuthResponse {
    private Long id;
    private String name;
    private String email;
    private String role;
    private String message;

    public Long getId() { return id; }
    public String getName() { return name; }
    public String getEmail() { return email; }
    public String getRole() { return role; }
    public String getMessage() { return message; }
}

File: app_flutter/lib/models/auth_response.dart
class AuthResponse {
  final int? id;
  final String? name;
  final String? email;
  final String? role;
  final String? message;

  AuthResponse({this.id, this.name, this.email, this.role, this.message});

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
    id: json['id'] as int?,
    name: json['name'] as String?,
    email: json['email'] as String?,
    role: json['role'] as String?,
    message: json['message'] as String?,
  );
}
